src/Catel.Analyzers/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -26,8 +27,16 @@ public static bool HasCallersInTypeAsync(this IMethodSymbol methodSymbol, Cancel
             try
             {
                 var methodName = methodSymbol.Name;
-                node = classDeclaration.DescendantNodes().Cast<InvocationExpressionSyntax>()
-                    .FirstOrDefault(x => string.Equals(((MemberAccessExpressionSyntax)x.Expression).Name.ToString(), methodName));
+                node = classDeclaration.DescendantNodes().OfType<InvocationExpressionSyntax>()
```

Comment:
if we don't need the node when checking the reference, maybe we should check reference first, and then the node?